### PR TITLE
my budget app final touches

### PR DESCRIPTION
**Pull request by tamana-Balkhi**
# Hi dear code reviewer 
# in this pull request I have implemented the project according to requirement

# Project requirements
 ### Design

-  You should follow these [design guidelines](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), including:
- Colors.
- Typography: font face, size and weight.
- Layout: composition and space between elements.

### NOTE: In these design guidelines there are several UIs that you won't need for this exercise; also, some pages are not given a design and you will create them following the design guidelines of the other pages.
Original design idea by [Gregoire Vella on Behance](https://www.behance.net/gregoirevella).

The [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/) requires that you give appropriate credit to the author. Therefore, you must do it in the README of your project.

> Interactions

> Splash screen

- A simple page with the name of your app (yes, you need to choose one), and links to the signup and log-in pages.
- Sign up and login pages
- The user should be able to register in the app with full name, email, and password (all mandatory).
- The user can log into the app using email and password.
- If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).
Home page (categories page)

> When the user logs in, they are presented with the categories page.

- For each category, the user can see their name, icon and the total amount of all the transactions that belong to that category.
- When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
- There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

> Transactions page

- For a given category, the list of transactions is presented, ordered by the most recent.
- At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
- There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
- When the user clicks on the "Back" button (<), the user navigates to the home page.

> "Add a new category" page

- The user fills out a form to create a new category, indicating their name and icon (both mandatory).
- The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.
- When the user clicks on the "Back" button (<), the user navigates to the home page.

> "Add a new transaction" page

- The user fills out a form to create a new transaction with:
- name (mandatory)
- amount (mandatory)
- categories (mandatory at least one)
- The user click (or taps) the "Save" button to create the new transation, and is taken to the transactions page for that category.
- When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,187 @@
...
-This README would normally document whatever steps are necessary to get the
-application up and running.
+# 📗 Table of Contents
 
-Things you may want to cover:
+- [📖 About the Project](#about-project)
+  - [🛠 Built With](#built-with)
+    - [Tech Stack](#tech-stack)
+    - [Key Features](#key-features)
+  - [🚀 Live Demo](#live-demo)
```
Comment (Deepakdanger) [inline README.md:8]: - [ ] A link to the deployed project should be present in the README.

File: app/views/entities/_entity.html.erb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<div id="<%= dom_id entity %>" class="entity">
+  <p>
+    <strong><%= entity.name %></strong>
+  </p>
+
+  <p>
+    <strong>$ <%= entity.amount %></strong>
+  </p>
+
```
Comment (Deepakdanger) [inline app/views/entities/_entity.html.erb]: - [ ] There should be a date mentioned when the transaction was created as per design. Please add in the UI.

### Your design
Screenshot from 2023-03-26 00-57-45


### Given design

Screenshot from 2023-03-26 00-58-10

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @tamana-Balkhi, 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 

## Required changes
_Check the comments under the review._


## Optional changes
You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: app/views/entities/_entity.html.erb (hunk 2 above)
Comment (tamana-Balkhi) [reply app/views/entities/_entity.html.erb]: thanks to mention, it is fixed now

**Comment by tamana-Balkhi** [comment]
greate

**Comment by tamana-Balkhi** [comment]
## dear code reviewer, 
please approve my pull request I tried a lot  and there is not any issue with my project  I have no more time today to bring changes on 

If I couldn't submit the project it will cause the model to repeat  please ## approve 

if problem is on deploying i am still working but its not working

**Review by Reem-lab — APPROVED** [review]
# Hi @tamana-Balkhi,👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🔥🔥💯💯


## Highlights 🔥🔥🔥🔥
- Followed Gitflow ✅
- No Linters error 🤩🔥💯
- Great working on the functionality🔥🔥
- Professional README 🔥🔥
- Great working on the design 🚀
- Good working on the presentation video 💯💯



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**